template/src/App.tsx

```diff
@@ -20,14 +23,38 @@ interface State {
   // client: ApolloClient<InMemoryCache>;
 }
 
+// TODO(ratkinson): convert to FC
 class App extends Component<{}, State> {
   public state = {};
 
-  public async componentDidMount() {
+  private setupApp = (): void => {
+    try {
+      // Use this block to setup either the store or the apollo client
+      // const rootStore = await setupRootStore();
+      // this.setState({ store });
+      //
+      // const client = await setupApolloClient();
+      // this.setState({ client })
+    } catch (error) {
+      throw error;
+    } finally {
+      // Now that everything is hydrated, hide the splashscreen
+      SplashScreen.hide();
+    }
+  };
+
+  public componentDidMount = async (): Promise<void> => {
     this.setupApp();
-  }
+  };
+
+  public renderNavigation = (): ReactElement => {
+    // Generate the root nav of the app based on presence of a user (restored in setup functions above)
+    const currentUser = false;
+
+    return currentUser ? <AppNav /> : <GuestNav />;
```

Comment:
Note that the switchNavigator was deprecated from react-navigation since this is a viable pattern with the new react-navigation component-based architecture.